specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/examples/Patches_Get.json

```diff
@@ -0,0 +1,35 @@
+{
+  "operationId": "Patcges_Get",
+  "title": "Patches_Get_0",
+  "parameters": {
+    "api-version": "2023-05-02-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "rg",
+    "builderName": "testBuilder",
+    "buildName": "testPatch"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg/providers/Microsoft.App/builders/testBuilder/patches/testPatch",
+        "name": "testPatch",
+        "type": "Microsoft.App/builders/patches",
+        "properties": {
+          "provisioningState": "Succeeded",
+          "patchStatus": "InProgress",
+          "configuration": {
+            "patchId": "1ce4b2a0"
```

Comment:
Some initial questions regarding the patch object:
- Are these the only contents that we want to expose to the customer? 
- What is the patch ID used for and how does it differ from the patch name? 
- How do we expect customers to decide whether or not to apply a patch? It seems like it could be useful to be able to include some kind of summary of what the patch will entail? Like old image / new image or an explanation of the patch, e.g. "Updates dotnet 7.0.1 to dotnet 7.0.2 by updating {old image} to {new image}." I think we just need to consider the user experience that we want, and how this can be used by the CLI and portal

Reply:
For this one, user is expected to pass in the patch id in order to apply specific patches. When they uses `patch list` command, it should return a list looks like this:
```
"1ce4b2a0":
{
    "id": "1ce4b2a0",
    "newRunImage": "run-dotnet-aspnet-7.0.5",
    "oldRunImage": "run-dotnet-aspnet-7.0.1",
    "reason": "New DotNet run image is now available.",
    "targetResourceGroup": "test-rg",
    "targetContainerAppEnvironmentName": "test-env",
    "targetContainerAppName": "test-app",
    "TargetContainerName": "test-container",
    "targetImageName": "testacr.azurecr.io/repo:before-upate"
}
``` 
After the user applies the patch using `patch apply 1ce4b2a0`, it should return something like this:
``` 
Patch applied.
{
    "newImageName": "testacr.azurecr.io/repo:before-update-patched-<timestamp>",
    "revisionInfo":
...
}
```

Reply:
yes, i'm not sure if this one sounds like a good idea or not, but the idea of the GET request is to directly fetch from the db. Each patcher in a managed env has its related patches saved in the db. The GET request will go to the db and fetch all of them and return them to the user when they run `patch list`